File: frontend/src/lib/exportV2.ts
const BASE = "/api/export-v2";

export interface AssetExportPackage {
  id: string;
  asset_id: string;
  asset_name: string;
  export_type: string;
  version_exported: number;
  version_label: string;
  included_files: string[];
  manifest_path: string;
  zip_path: string;
  zip_size: number;
  normalized: boolean;
  has_textures: boolean;
  has_inspection: boolean;
  has_thumbnail: boolean;
  created_at: string;
}

export const EXPORT_TYPES = [
  {
    value: "glb_package",
    label: "GLB Package",
    description: "Model file + manifest. Clean minimal delivery.",
    includes: ["model", "thumbnail", "manifest"],
  },
  {
    value: "game_ready",
    label: "Game Ready",
    description: "Model + all project textures + thumbnail + manifest.",
    includes: ["model", "textures", "thumbnail", "manifest"],
  },
  {
    value: "texture_bundle",
    label: "Texture Bundle",
    description: "Project textures only + manifest.",
    includes: ["textures", "manifest"],
  },
  {
    value: "inspection_bundle",
    label: "Inspection Bundle",
    description: "Model + inspection report + manifest.",
    includes: ["model", "inspection", "manifest"],
  },
  {
    value: "full_project_bundle",
    label: "Full Project Bundle",
    description: "All available files — model, textures, thumbnail, inspection.",
    includes: ["model", "textures", "thumbnail", "inspection", "manifest"],
  },
] as const;

export const VERSION_OPTIONS = [
  { value: "latest", label: "Latest Version", description: "Current version of the asset" },
  { value: "original", label: "Original", description: "First imported/registered version" },
  { value: "normalized", label: "Normalized", description: "Blender-normalized version (if available)" },
] as const;

export function formatPackageSize(bytes: number): string {
  if (bytes === 0) return "0 B";
  if (bytes < 1024) return `${bytes} B`;
  if (bytes < 1024 * 1024) return `${(bytes / 1024).toFixed(1)} KB`;
  return `${(bytes / (1024 * 1024)).toFixed(2)} MB`;
}

export async function createExportPackage(
  assetId: string,
  exportType: string,
  versionLabel: string,
): Promise<AssetExportPackage> {
  const res = await fetch(`${BASE}/create`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ asset_id: assetId, export_type: exportType, version_label: versionLabel }),
  });
  if (!res.ok) {
    const body = await res.json().catch(() => ({}));
    throw new Error(body.detail ?? `Create export package failed (${res.status})`);
  }
  return res.json();
}

export async function getExportPackage(packageId: string): Promise<AssetExportPackage> {
  const res = await fetch(`${BASE}/${packageId}`);
  if (!res.ok) throw new Error(`Get export package failed (${res.status})`);
  return res.json();
}

export async function listExportPackages(assetId?: string): Promise<AssetExportPackage[]> {
  const url = assetId ? `${BASE}?asset_id=${assetId}` : BASE;
  const res = await fetch(url);
  if (!res.ok) throw new Error(`List export packages failed (${res.status})`);
  return res.json();
}

export function getPackageDownloadUrl(packageId: string): string {
  return `${BASE}/${packageId}/download`;
}

File: backend/app/models/export_v2.py
from __future__ import annotations
from pydantic import BaseModel

VALID_EXPORT_TYPES = {
    "glb_package",
    "game_ready",
    "texture_bundle",
    "inspection_bundle",
    "full_project_bundle",
}

# Which file categories each export type includes
EXPORT_TYPE_FLAGS: dict[str, dict[str, bool]] = {
    "glb_package":         {"model": True,  "textures": False, "inspection": False, "thumbnail": True},
    "game_ready":          {"model": True,  "textures": True,  "inspection": False, "thumbnail": True},
    "texture_bundle":      {"model": False, "textures": True,  "inspection": False, "thumbnail": False},
    "inspection_bundle":   {"model": True,  "textures": False, "inspection": True,  "thumbnail": False},
    "full_project_bundle": {"model": True,  "textures": True,  "inspection": True,  "thumbnail": True},
}


class ExportPackageRequest(BaseModel):
    asset_id: str
    export_type: str = "full_project_bundle"
    version_label: str = "latest"   # "latest" | "original" | "normalized"


class ExportManifest(BaseModel):
    minimesh_version: str = "2"
    asset_id: str
    asset_name: str
    version: int
    version_label: str
    provider: str
    normalized: bool
    export_type: str
    textures: list[dict] = []
    has_inspection: bool = False
    inspection_summary: dict | None = None
    thumbnail: str | None = None
    file_count: int = 0
    exported_at: str


class AssetExportPackage(BaseModel):
    id: str
    asset_id: str
    asset_name: str
    export_type: str
    version_exported: int = 1
    version_label: str = "latest"
    included_files: list[str] = []
    manifest_path: str = ""
    zip_path: str = ""
    zip_size: int = 0
    normalized: bool = False
    has_textures: bool = False
    has_inspection: bool = False
    has_thumbnail: bool = False
    created_at: str
